source/core/DualTypeVisitor.cpp

```diff
@@ -188,19 +176,23 @@ namespace Vireo
     }
 
     //------------------------------------------------------------
-    bool DualTypeVisitor::ArrayCompatible(TypeRef typeRefX, TypeRef typeRefY, DualTypeOperation* operation)
+    bool DualTypeVisitor::ArrayCompatible(TypeRef typeRefX, void* pDataX, TypeRef typeRefY, void* pDataY, const DualTypeOperation &operation)
     {
         // Verify number of dimensions are the same
         bool success = typeRefX->Rank() == typeRefY->Rank();
         // Verify each dimension has the same size
         if (success) {
-            TypedArrayCoreRef arrayX = *(static_cast<const TypedArrayCoreRef*>(typeRefX->Begin(kPARead)));
-            TypedArrayCoreRef arrayY = *(static_cast<const TypedArrayCoreRef*>(typeRefY->Begin(kPARead)));
+            TypedArrayCoreRef arrayX = *(static_cast<const TypedArrayCoreRef*>(pDataX));
+            TypedArrayCoreRef arrayY = *(static_cast<const TypedArrayCoreRef*>(pDataY));
+            if (operation.ShouldInflateDestination()) {
+                arrayY->ResizeToMatchOrEmpty(arrayX);
+            }
             IntIndex* dimensionLenghtsX = arrayX->DimensionLengths();
```

Comment:
Looks like I missed during the first pass. Apologies.
The while loop to check for dimensions may be done only if ShouldInflateDestination() returns false (as an else case), as the inflation case anyhow resizes to match.

Reply:
Fair enough. Will move that to the else case.